datalake/src/main/java/com/sequenceiq/datalake/flow/SdxReactorFlowManager.java

```diff
@@ -115,6 +132,16 @@ public FlowIdentifier triggerSdxResize(Long sdxClusterId, SdxCluster newSdxClust
         boolean performRestore = sdxBackupRestoreService.shouldSdxRestoreBePerformed(
                 newSdxCluster, entitlementService.isDatalakeBackupOnResizeEnabled(ThreadBasedUserCrnProvider.getAccountId())
         );
+        DatalakeBackupInfo lastSuccessfulBackup = datalakeDrClient.getLastSuccessfulBackup(newSdxCluster.getClusterName(), userId, Optional.empty());
```

Comment:
Consider moving this logic to the backup restore service and exposing a method. That way we can avoid the new injection and keep this method as simple as possible.